File: src/main/java/com/wesleyhome/test/jupiter/InvalidParameterException.java
package com.wesleyhome.test.jupiter;

import java.util.List;
import java.util.stream.Collectors;

@SuppressWarnings("ALL")
public class InvalidParameterException extends RuntimeException {

  public InvalidParameterException(Class<?> parameterClass, List<Class<?>> dataProviderClasses) {
    super(String.format("Unable to find suitable data provider for %s from %s",
      parameterClass.getSimpleName(),
      dataProviderClasses.stream()
        .map(Class::toString)
        .collect(Collectors.joining(", "))
    ));
  }
}

File: extension/src/main/kotlin/com/wesleyhome/test/jupiter/InvalidParameterException.kt
package com.wesleyhome.test.jupiter

import com.wesleyhome.test.jupiter.provider.TestParameter

class InvalidParameterException(testParameter: TestParameter) :
    RuntimeException("Unable to find a suitable data provider for parameter [${testParameter.name}] with type '${testParameter.type.qualifiedName}'")
